draft/spec/inventory_schema.json

```diff
@@ -34,44 +34,44 @@
             "enum": [ "Object" ]
         },
         "versions": {
-            "type" : "array",
-            "uniqueItems": true,
-            "items": {
-                "type": "object",
-                "additionalProperties": false,
-                "properties": {
-                    "created": {
-                        "type": "string",
-                        "format": "date-time"
-                    },
-                    "message": {
-                        "type": "string"
-                    },
-                    "state": {
-                        "$ref": "#/definitions/digests_to_files"
-                    },
-                    "type": {
-                        "description": "Seems that using `const` would be nicer but that doesn't seem to work with Python jsonschema",
-                        "enum": [ "Version" ]
-                    },
-                    "user": {
-                        "type": "object",
-                        "additionalProperties": false,
-                        "properties": {
-                            "address": {
-                                "type": "string"
-                            },
-                            "name": {
-                                "type": "string"
-                            }
+            "description": "Each key is a version directory name with version objects as values",
+            "type" : "object",
+            "additionalProperties": false,
+            "patternProperties": {
+                "^v\\d+$": {
```

Comment:
Can we use this regex to forbid non-zero initial numbers?

Reply:
No, that would be wrong. v99999 is allowed if you start with v1

Reply:
...and v09999 is right if you start with v00001

Reply:
Hm. So how will we validate:

"If zero-padded version directory numbers are used then they must start with the prefix v and then a zero. For example, in an implementation that uses five digits for version directory names then v00001 to v09999 are allowed, v10000 is not allowed."

Reply:
With code rather than JSON schema. There are lots of parts of the spec than cannot be validated with the schema (like the checksum actually being right). There may be some more complex JSON schema foo that would get us a bit further, but it certainly won't get us all the way